### DISCUSSION TITLE
Use use-context-selector for performance improvements?

### DISCUSSION THREAD

**Discussion by pserrer1** [react-hook-form/react-hook-form#8096] [Ideas] [answered]
Hey,

are there any plans to use use-context-selector instead of native context to improve the performance?
We have deeply nested form and I assume this could help.

**Comment by Moshyfawn** [comment, marked as answer]
I can't say for sure. @bluebill1049 can elaborate further. But RHF strives to be the deps free lib, so I doubt it's gonna make use of [the use-context-selector lib](https://github.com/dai-shi/use-context-selector) at this time.

With [React 18](https://github.com/reactjs/rfcs/blob/react-18/text/0000-react-18.md) coming, there's a chance that the type of optimization [the use-context-selector lib](https://github.com/dai-shi/use-context-selector) brings wouldn't be needed. At least in RHF case.
